readthedocs/projects/forms.py

```diff
@@ -176,6 +176,22 @@ class Meta(object):
     )
 
 
+class ListTextWidget(forms.TextInput):
+    def __init__(self, data_list, name, *args, **kwargs):
+        super(ListTextWidget, self).__init__(*args, **kwargs)
+        self._name = name
+        self._list = data_list
+        self.attrs.update({'list': 'list__%s' % self._name})
+
+    def render(self, name, value, attrs=None):
+        text_html = super(ListTextWidget, self).render(name, value, attrs=attrs)
+        data_list = '<datalist id="list__%s">' % self._name
```

Comment:
What's the difference between a `datalist` and a simple `select` html tag?

Seems that `data list` is not supported in Safari, https://www.w3schools.com/tags/tag_datalist.asp